### PR TITLE
DATAUP-389 - add run_job integration test

### PR DESCRIPTION
**Pull request by MrCreosote**
# Description of PR purpose/changes

What it says on the tin. Note that the notes at the top of the `api_to_db_test.py` file are pretty important.

# Jira Ticket / Github Issue #
- [x] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [x] Tests pass in Github Actions and locally 
- [x] Changes available by spinning up a local test suite

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [n/a] I have made corresponding changes to the documentation
- [?] My changes generate no new warnings - 3k+ warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [x] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the Github Actions build passes)

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: test/tests_for_integration/api_to_db_test.py [hunk 1]
```diff
@@ -287,7 +368,238 @@ def test_get_admin_permission_success(ee2_port):
...
+    wsc.create_workspace({"workspace": "foo"})
+    wsc.save_objects(
+        {
+            "id": 1,
+            "objects": [
+                {"name": "one", "type": "Trivial.Object-1.0", "data": {}},
+                {"name": "two", "type": "Trivial.Object-1.0", "data": {}},
+            ],
+        }
+    )
```
Comment (MrCreosote) [inline test/tests_for_integration/api_to_db_test.py:412]: In a later PR I'll factor out the workspace setup, the condor expected sub creation, and the mongo job checking into functions for reuse with other tests in this module. Already done locally.

File: test/tests_for_integration/api_to_db_test.py [hunk 2]
```diff
@@ -287,7 +368,238 @@ def test_get_admin_permission_success(ee2_port):
...
+    err = (
+        "('An error occurred while fetching user permissions from the Workspace', "
+        + "ServerError('No workspace with id 1 exists'))"
+    )
+    _run_job_fail(ee2_port, TOKEN_NO_ADMIN, params, err)
+
+
+def test_run_job_fail_bad_method(ee2_port):
+    params = {"method": "mod.meth.moke", "app_id": "mod/app"}
+    # TODO the Server.py file is quoting strings for some reason it seems
```
Comment (MrCreosote) [inline test/tests_for_integration/api_to_db_test.py:553]: I've got this fixed locally, will PR in the future.

File: test/utils_shared/test_utils.py [hunk 3]
```diff
@@ -393,6 +394,15 @@ def assert_exception_correct(got: Exception, expected: Exception):
     assert type(got) == type(expected)
 
 
+def assert_close_to_now(time_):
```
Comment (ialarmedalien) [inline test/utils_shared/test_utils.py:397]: slightly odd variable name -- I'd expect `_time` or perhaps `time_to_test`. The hanging underscore is unexpected!

Comment (MrCreosote) [reply test/utils_shared/test_utils.py:397]: Suffixing a keyword with an underscore is actually a standard convention in python:
https://www.datacamp.com/community/tutorials/role-underscore-python#STU
https://www.geeksforgeeks.org/underscore-_-python/

**Review by ialarmedalien — APPROVED** [review]
...

File: test/utils_shared/test_utils.py (hunk 3 above)
Comment (dakotablair) [reply test/utils_shared/test_utils.py:397]: Mostly I have seen it used, as it is in this case, to distinguish the variable from the corresponding standard library object or keyword.

**Pull request merged**